community/modules/scheduler/htcondor-central-manager/main.tf

```diff
@@ -0,0 +1,169 @@
+}
+
+data "google_compute_image" "htcondor" {
+  family  = var.instance_image.family
+  project = var.instance_image.project
+
+  lifecycle {
+    postcondition {
+      condition     = self.disk_size_gb <= var.disk_size_gb
+      error_message = "var.disk_size_gb must be set to at least the size of the image (${self.disk_size_gb})"
+    }
+  }
+}
+
+data "google_compute_zones" "available" {
+  project = var.project_id
+  region  = var.region
+}
+
+data "google_compute_region_instance_group" "cm" {
+  self_link = time_sleep.mig_warmup.triggers.self_link
+}
+
+data "google_compute_instance" "cm" {
+  self_link = data.google_compute_region_instance_group.cm.instances[0].instance
+}
+
+resource "google_storage_bucket_object" "cm_config" {
+  name    = "${var.deployment_name}-cm-config-${substr(md5(local.cm_config), 0, 4)}"
+  content = local.cm_config
+  bucket  = var.htcondor_bucket_name
+}
+
+module "startup_script" {
+  source = "github.com/GoogleCloudPlatform/hpc-toolkit//modules/scripts/startup-script?ref=v1.20.0&depth=1"
+
+  project_id      = var.project_id
+  region          = var.region
+  labels          = local.labels
+  deployment_name = var.deployment_name
+
+  runners = local.all_runners
+}
+
+module "central_manager_instance_template" {
+  # tflint-ignore: terraform_module_pinned_source
+  source = "github.com/terraform-google-modules/terraform-google-vm//modules/instance_template?ref=84d7959"
+
+  name_prefix = local.name_prefix
+  project_id  = var.project_id
+  network     = var.network_self_link
+  subnetwork  = var.subnetwork_self_link
+  service_account = {
+    email  = var.central_manager_service_account_email
+    scopes = var.service_account_scopes
+  }
+  labels = local.labels
+
+  machine_type   = var.machine_type
+  disk_size_gb   = var.disk_size_gb
+  preemptible    = false
+  startup_script = module.startup_script.startup_script
+  metadata       = local.metadata
+  source_image   = data.google_compute_image.htcondor.self_link
+}
+
+module "htcondor_cm" {
+  # tflint-ignore: terraform_module_pinned_source
+  source = "github.com/terraform-google-modules/terraform-google-vm//modules/mig?ref=84d7959"
+
+  project_id        = var.project_id
+  region            = var.region
+  target_size       = 1
+  hostname          = local.name_prefix
+  instance_template = module.central_manager_instance_template.self_link
+
+  health_check_name = "health-${local.name_prefix}"
+  health_check = {
+    type                = "tcp"
+    initial_delay_sec   = 600
+    check_interval_sec  = 20
+    healthy_threshold   = 2
+    timeout_sec         = 8
+    unhealthy_threshold = 3
+    response            = ""
+    proxy_header        = "NONE"
+    port                = 9618
+    request             = ""
+    request_path        = ""
+    host                = ""
+    enable_logging      = true
+  }
+
+  update_policy = [{
```

Comment:
I think it would be appropriate to document somewhere in layman's terms what the HA characteristics are.

Reply:
PTAL